config/confighttp/clientinfohandler.go

```diff
@@ -46,6 +46,8 @@ func contextWithClient(req *http.Request) context.Context {
 		cl.Addr = ip
 	}
 
+	cl.Metadata = req.Header
```

Comment:
I think you need to "clone" the headers, otherwise it is not guaranteed that they are not changed/modified after the Handler func finishes. Because of that I think we should have this configured by the user.

Reply:
@bogdandrutu Makes sense, but I'm not sure how to make this configurable. Configurable `clientInfoHandler`? It'll eventually circle back to an auth-like extension in the original contrib PR.